pkg/manager/member/tidbcluster_status_manager.go

```diff
@@ -105,7 +105,18 @@ func (tcsm *TidbClusterStatusManager) syncTidbMonitorRef(tc *v1alpha1.TidbCluste
 }
 
 func (tcsm *TidbClusterStatusManager) syncDashboardMetricStorage(tc *v1alpha1.TidbCluster, tm *v1alpha1.TidbMonitor) error {
-	pdEtcdClient, err := tcsm.pdControl.GetPDEtcdClient(pdapi.Namespace(tc.Namespace), tc.Name, tc.IsTLSClusterEnabled())
+	if tc.Spec.PD == nil {
+		return nil
+	}
+
+	var pdEtcdClient pdapi.PDEtcdClient
+	var err error
+	if tc.IsHeterogeneous() {
+		pdEtcdClient, err = tcsm.pdControl.GetPDEtcdClient(pdapi.Namespace(tc.Namespace), tc.Spec.Cluster.Name, tc.IsTLSClusterEnabled())
+	} else {
+		pdEtcdClient, err = tcsm.pdControl.GetPDEtcdClient(pdapi.Namespace(tc.Namespace), tc.Name, tc.IsTLSClusterEnabled())
+	}
+
```

Comment:
Here `isHeterogeneous` will never be true.

Reply:
I think when pd support clusterRef ,this code will  work.

Reply:
`IsHeterogeneous ` checks `tc.spec.pd==nil`.